addons/sourcemod/scripting/Spectate.sp

```diff
@@ -391,7 +379,14 @@ public Action Command_Spectate(int client, int argc)
 	#endif
 	}
 
-	ChangeClientTeam(client, CS_TEAM_SPECTATOR);
+	if (!g_cEnable.BoolValue && IsNotSpectator)
```

Comment:
put it before `if (g_cSpecLimit.IntValue > 0 && IsNotSpectator)`

Reply:
It's done on purpose, to allow the usage of the command for player who are already in spec team, like that they can use the command to switch faster to spectate someone.
But I can refactor a bit.